postreise/plot/plot_states.py

```diff
@@ -65,109 +65,156 @@ def get_state_borders():
         us_states_dat = get_state_borders()
+    if state_list is None:
+        # Default to continental US from state borders data, w/ no colors, no labels.
+        if labels_list is not None:
+            raise TypeError("Cannot specify labels_list without state_list")
+        if colors is not None and not isinstance(colors, str):
+            raise TypeError("Cannot specify list of colors without state_list")
+        state_list = list(set(us_states_dat.keys()) - {"AK", "HI", "DC", "PR"})
+    else:
+        if colors is not None:
+            if not isinstance(colors, str):
+                try:
+                    if len(state_list) != len(colors):
+                        raise ValueError("state_list and colors must be same length")
+                except TypeError:
+                    raise TypeError("colors must be str or iterable")
+        if labels_list is not None:
+            try:
+                if len(state_list) != len(labels_list):
+                    raise ValueError("State_list and labels_list must be same length")
+            except TypeError:
+                raise TypeError("labels_list must be an iterable")
+    if legend_colors is not None or legend_labels is not None:
+        try:
+            if len(legend_colors) != len(legend_labels):
+                raise ValueError("Length of legend_colors and legend_labels must match")
+        except TypeError:
+            raise TypeError("legend_colors and legend_labels must be iterables or None")
+    # If no color specified, default to white. Either way, apply to all states.
+    colors = "white" if colors is None else colors
+    colors = [colors for s in state_list] if isinstance(colors, str) else colors
     all_state_xs, all_state_ys = project_borders(us_states_dat, state_list=state_list)
-    # Initialize figure
-    p = figure(
-        tools="pan,wheel_zoom,reset,save",
-        x_axis_location=None,
-        y_axis_location=None,
-        plot_width=800,
-        plot_height=800,
-        output_backend="webgl",
-        sizing_mode="stretch_both",
-        match_aspect=True,
-    )
-    # legend squares, plot behind graph so hidden
-    for i in range(len(leg_color)):
-        p.square(
-            -8.1e6,
-            [5.2e6, 5.3e6],
-            fill_color=leg_color[i],
-            color=leg_color[i],
-            size=300,
-            legend_label=leg_label[i],
+    # Initialize figure if necessary
+    if bokeh_figure is None:
+        p = figure(
+            tools="pan,wheel_zoom,reset,save",
+            x_axis_location=None,
+            y_axis_location=None,
+            plot_width=800,
+            plot_height=800,
+            output_backend="webgl",
+            sizing_mode="stretch_both",
+            match_aspect=True,
         )
+    else:
+        p = bokeh_figure
+    # if legend info is passed, plot legend squares behind graph so they're hidden
+    if legend_colors is not None:
+        for i, _ in enumerate(legend_colors):
```

Comment:
Why don't you assign the current color to a variable and use it in the loop in place of `legend_colors[i]`. Just curious

Reply:
I was adapting old code that used this code pattern, but I agree that your suggestion is cleaner. I'll change it.

Reply:
Done